src/uct/api/uct_def.h

```diff
@@ -210,6 +249,149 @@ typedef struct uct_cm_remote_data {
+
+    /** Enables @ref uct_cm_listener_conn_request_args::remote_data
+     *  Indicates that remote_data field in uct_cm_listener_conn_request_args_t
+     *  is valid.
+     */
+    UCT_CM_LISTENER_CONN_REQUEST_ARGS_FIELD_REMOTE_DATA  = UCS_BIT(2)
+};
+
+
+/**
+ * @ingroup UCT_CLIENT_SERVER
+ * @brief Arguments to the listener's connection request callback.
+ *
+ * The local device name, connection request handle and the data the client sent.
+ * Used with the client-server API on a connection manager.
+ */
+typedef struct uct_cm_listener_conn_request_args {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref uct_cm_listener_conn_request_args_field.
+     * Fields not specified by this mask will be ignored.
+     */
+    uint64_t                   field_mask;
+
+    /**
+     * Local device name which handles the incoming connection request.
+     */
+    char                       dev_name[UCT_DEVICE_NAME_MAX];
+
+    /**
+     * Connection request handle. Can be passed to this callback from the
+     * transport and will be used by it to accept or reject the connection
+     * request from the client.
+     */
+    uct_conn_request_h         conn_request;
+
+    /**
+     * Remote data from the client.
+     */
+    const uct_cm_remote_data_t *remote_data;
+} uct_cm_listener_conn_request_args_t;
+
+
+/**
+ * @ingroup UCT_CLIENT_SERVER
+ * @brief Field mask flags for client-side connection established callback.
+ *
+ * The enumeration allows specifying which fields in
+ * @ref uct_cm_ep_client_connect_args are present, for backward compatibility
+ * support.
+ */
+enum uct_cm_ep_client_connect_args_field {
+    /** Enables @ref uct_cm_ep_client_connect_args::remote_data */
+    UCT_CM_EP_CLIENT_CONNECT_ARGS_FIELD_REMOTE_DATA = UCS_BIT(0),
+
+    /** Enables @ref uct_cm_ep_client_connect_args::status */
+    UCT_CM_EP_CLIENT_CONNECT_ARGS_FIELD_STATUS      = UCS_BIT(1)
+};
+
+
+/**
+ * @ingroup UCT_CLIENT_SERVER
+ * @brief Arguments to the client's connect callback.
+ *
+ * Used with the client-server API on a connection manager.
+ */
+typedef struct uct_cm_ep_client_connect_args {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref uct_cm_ep_server_connect_args_field.
+     * Fields not specified by this mask will be ignored.
+     */
+    uint64_t                   field_mask;
+
+    /**
+     * Remote data from the server.
+     */
+    const uct_cm_remote_data_t *remote_data;
+
+    /**
+     * Indicates the server's status.
+     */
+    ucs_status_t               status;
+} uct_cm_ep_client_connect_args_t;
+
+
+/**
+ * @ingroup UCT_CLIENT_SERVER
+ * @brief Field mask flags for server-side connection established callback.
+ *
+ * The enumeration allows specifying which fields in
+ * @ref uct_cm_ep_server_connect_args are present, for backward compatibility
+ * support.
+ */
+enum uct_cm_ep_server_connect_args_field {
```

Comment:
didn't we decide to name it server_notify ?
e.g uct_cm_ep_server_notify_args_field

Reply:
I thought notify was decided for the new api, that isn't present in this pr (when client would notify the server that it's connected)

Reply:
right, and the names in this API have to match it as well

Reply:
but if the new api wasn't introduced yet, why change it now?
can it be done together with the new api?